src/adera/Shaders/PlumeShader.cpp

```diff
@@ -24,31 +24,53 @@
  */
 #include "PlumeShader.h"
 
+#include <osp/types.h>
+#include <osp/Active/SysRender.h>
+#include <osp/Resource/AssetImporter.h>
+
 #include <Magnum/GL/Version.h>
 #include <Magnum/GL/Shader.h>
 #include <Corrade/Containers/Reference.h>
 #include <Magnum/GL/Texture.h>
 
-using namespace Magnum;
+using Magnum::Int;
+
+using namespace osp;
 using namespace osp::active;
 using namespace adera::shader;
 
-void PlumeShader::draw_plume(ActiveEnt e, ActiveScene& rScene, GL::Mesh& rMesh,
-    ACompCamera const& camera, ACompTransform const& transform)
+void PlumeShader::draw_plume(ActiveEnt e,
+    ActiveScene& rScene, ACompCamera const& camera)
 {
-    auto& shaderInstance = rScene.reg_get<ACompPlumeShaderInstance>(e);
-    PlumeShader& shader = *shaderInstance.m_shaderProgram;
+    auto& resources = rScene.get_context_resources();
+    PlumeShader& shader = *resources.get<PlumeShader>(smc_resourceName);
+
+    using Magnum::GL::Texture2D;
+    using Magnum::GL::Mesh;
+    // Collect uniform data
+    auto& transform = rScene.reg_get<ACompTransform>(e);
+    auto& rPlumeComp = rScene.reg_get<ACompExhaustPlume>(e);
+    auto const& plumedata = *rPlumeComp.m_effect;
+    Mesh& rMesh = *rScene.reg_get<ACompMesh>(e).m_mesh;
+
+    constexpr std::string_view texName = "OSPData/adera/noise1024.png";
```

Comment:
no need to change it for this PR, but eventually we should make sure either:
1) Textures like this are directly imported into the binary, there are various techniques for how to do this, so i won't detail the mechanism that's used, but the point is that it's not loaded from disk.
2) Resources loaded from disk are described in a config file, so that they can be changed by the end user.

Reply:
This will all be config'd eventually, although importing into the binary is an interesting idea for loading shaders.